Mathlib/Data/Real/EReal.lean

```diff
@@ -800,6 +800,56 @@ theorem top_add_coe (x : ℝ) : (⊤ : EReal) + x = ⊤ :=
   rfl
 #align ereal.top_add_coe EReal.top_add_coe
 
+/--For any extended real number `x` which is not `⊥`, the sum of `⊤` and `x` is equal to `⊤`.-/
+@[simp]
+theorem top_add_of_ne_bot {x : EReal} (h : x ≠ ⊥) : ⊤ + x = ⊤ := by
+  induction x using EReal.rec
+  · exfalso; exact h (Eq.refl ⊥)
+  · exact top_add_coe _
+  · exact top_add_top
+
+/--For any extended real number `x`, the sum of `⊤` and `x` is equal to `⊤`
+if and only if `x` is not `⊥`.-/
```

Comment:
style nit. Also, you can leave these docstrings in, but I don't think they're adding anything meaningful. Maybe in the future consider whether reading the docstring is actually simpler than reading the statement. :shrug:
```suggestion
/-- For any extended real number `x`, the sum of `⊤` and `x` is equal to `⊤`
if and only if `x` is not `⊥`. -/
```

Reply:
Yes, you're totally right. Sorry for the pedantry.